### PR TITLE
Remove redundant assertion checks in HyperOpt Optimizer.

### PR DESCRIPTION
**Pull request by c-bata**
In this PR, I remove duplicated assertion checks in HyperOpt Optimizer.

### CODE REVIEW

**Comment by rdturnermtl** [comment]
What is the motivation for removing these assertions? Is there any downside to having this extra check?

File: bayesmark/builtin_opt/hyperopt_optimizer.py [hunk 1]
```diff
@@ -227,7 +227,6 @@ def observe(self, X, y):
             x_guess_ = HyperoptOptimizer.hashable_dict(x_guess)
             assert x_guess_ in self.trial_id_lookup, "Appears to be guess that did not originate from suggest"
 
-            assert x_guess_ in self.trial_id_lookup, "trial object not available in trial dict"
```
Comment (c-bata) [inline bayesmark/builtin_opt/hyperopt_optimizer.py:230]: @rdturnermtl the motivation of this pr is that this assertion is already checked at L228.

If trial_id_lookup does not contain `x_guess_` , L228 will raise AssertionError. In such a situation, L230 won't be evaluated.

File: example_opt_root/hyperopt_optimizer.py [hunk 2]
```diff
@@ -215,7 +215,6 @@ def observe(self, X, y):
             x_guess_ = HyperoptOptimizer.hashable_dict(x_guess)
             assert x_guess_ in self.trial_id_lookup, "Appears to be guess that did not originate from suggest"
 
-            assert x_guess_ in self.trial_id_lookup, "trial object not available in trial dict"
```
Comment (c-bata) [inline example_opt_root/hyperopt_optimizer.py:218]: ditto.

**Comment by rdturnermtl** [comment]
ok, I will run some more tests and then merge.

**Pull request merged**